### PR TITLE
"isXXX" builtins

### PR DESCRIPTION
**Pull request by arduano**
"isXXX" builtins

Builtins for testing whether a value is of a certain type, should be relatively simple

### CODE REVIEW

File: nixjs-rt/src/builtins.ts [hunk 1]
```diff
@@ -385,39 +387,75 @@ export function getBuiltins() {
     },
 
     isAttrs: (arg) => {
-      throw new Error("unimplemented");
+      if (arg instanceof Attrset) {
```
Comment (urbas) [inline nixjs-rt/src/builtins.ts:390]: Consider moving `_nixBoolFromJs` to `utils.js` and using it here:
```suggestion
      return nixBoolFromJs(arg instanceof Attrset);
```

**Review by urbas — APPROVED** [review]

File: nixjs-rt/src/builtins.ts (hunk 1 above)
Comment (arduano) [reply nixjs-rt/src/builtins.ts]: What about just `new NixBool(arg instanceof Attrset)`

Pushed it as that just then

Comment (arduano) [reply nixjs-rt/src/builtins.ts]: My reasoning is why have a `nixBoolFromJs` but not a `nixStringFromJs` and `nixNumberFromJs`, etc, it's much easier to just use constructor calls

Comment (urbas) [reply nixjs-rt/src/builtins.ts]: `nixBoolFromJs` is an optimisation. To avoid allocating new instances for every `true` and `false`.

**Review by urbas — APPROVED** [review]
I would vote to avoid allocations of `NixBool` but not too fussed.

**Comment by arduano** [comment]
That's a good point. I didn't think of that. I'll move it and add comments to make the reasoning more clear.

**Comment by arduano** [comment]
Ok I didn't end up moving it into utils, but I noticed there were places in the code that directly rely on these object references being singletons, so I'll enforce that in a future pr otherwise this causes bugs

**Pull request merged**